src/dmd/lexer.d

```diff
@@ -227,7 +227,7 @@ class Lexer
                         goto LendSkipFourSpaces;
                     p++;
                 }
-                while (*(cast(uint*)p) == 0x20202020) // ' ' == 0x20
+                while (p + 4 < end && *(cast(uint*)p) == 0x20202020) // ' ' == 0x20
```

Comment:
I'm honestly against doing this check without proper benchmarking. This thing got introduced in DMD with poor benchmarking and, AFAIK Clang/GCC lexers doesn't consider this. It triggers very edgy cases and can lead to catastrophic failures. We have made a lot of workarounds in the compiler to support this and I doubt it brings any positive benefit overall.

Reply:
We could use some performance tests in CI, ideally. Unfortunately getting reliable performance data is not so easy unless you run a big suite of things to compare performance with.

Reply:
I mean these benchmarks https://github.com/dlang/dmd/pull/11095 . We tested this with a codebase full of spaces. What about a codebase with tabs or a generated file without spaces at all? I think this should be studied. I don't see any heavily used lexer doing this.

Reply:
GCC has a large amount of inline asm for different architectures to scan for the next line, using SSE2, AVX (ignoring Arm, PPC, SystemZ, and others) to scan for `\t\n\\\?` across a (vector) buffer with one instruction. So it's not strictly true to say that no one is optimizing this.

Reply:
Ok, didn't know about it. I mean it makes sense to me, but I expected more results.